image/image.go

```diff
@@ -87,6 +88,18 @@ func validate(w walker, refs []string, out *log.Logger) error {
 		if err := m.validate(w); err != nil {
 			return err
 		}
+
+		ml, err := findManifestList(w, d)
+		if err == nil {
+			if err := ml.validate(w); err != nil {
+				return err
+			}
+		} else {
+			if !strings.Contains(string(err), "manifestlist not found") {
+				return err
+			}
+		}
+
 		if out != nil {
```

Comment:
Should we `findManifestList` first and if manifest_list exists, then extract the manifest from the manifest_list and then validate the manifest?

I +1 with @wking 's proposal 
>I'd rather land opencontainers/image-spec#403 and use:

>Get an initial descriptor.
Fetch the object referenced by the descriptor.
Run the appropriate intra-blob validation on it (opencontainers/image-spec#403).
For any children referenced from the blob, recurse to (2).